datafusion/physical-expr/src/aggregate/first_last.rs

```diff
@@ -237,13 +231,17 @@ impl Accumulator for FirstValueAccumulator {
         };
         if !ordered_states[0].is_empty() {
             let first_row = get_row_at_idx(&ordered_states, 0)?;
-            let first_ordering = &first_row[1..];
+            // When collecting orderings, we exclude the is_set flag from the state.
+            let first_ordering = &first_row[1..is_set_idx];
```

Comment:
`is_set_idx` is an index for `states` but does `first_row` have the same index? Can they be different because `filtered_states` may have filtered rows?

Reply:
Please never mind, this should be the same

Reply:
Doesn't filtered_states come from `states`? Why they have different index?? They are simply partial aggregation inputs. Do you think there are different rows in partial aggregation inputs?